File: LeetCode/Medium/Container With Most Water/Java/containerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length-1;
        int res = 0;
        while (i<j) {
            int area = Math.min(height[i], height[j]) * (j-i);
            res = Math.max(res, area);
            if (height[i] < height[j]) i++;
            else j--;
        }
        return res;
    }
}

File: LeetCode/Medium/Container With Most Water/C++/container_with_most_water.cc
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0, j=height.size()-1;
        int res = 0;
        while(i < j) {
            int area = min(height[i], height[j]) * (j-i);
            res = max(res, area);
            if (height[i] > height[j]) j--;
            else i++;
        }
        return res;
    }
};
